### PR TITLE
Updated Organisation Tests with SeederData

### PR DESCRIPTION
**Pull request by iAKM**

### CODE REVIEW

File: src/__tests__/organization.tests.ts [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+
+    });
+
+    it("When creating new oraganization with same name, Expect to return HTTP Status 400", async () => {
+      //Arrange
+      let oraganizationApi = new OrganizationApi(configurationWithAuthHeader);
+      //Act
+      let payload = TdeiObjectFaker.getOrganization();
+      //TODO: correct it to name instead of id
+      payload.org_name = <string>seederData?.organizationId
```
Comment (MashB) [inline src/__tests__/organization.tests.ts]: Please look at the implementation in pathways for similar case

File: src/__tests__/organization.tests.ts [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+    
+          //Assert
+          await expect(oraganizationResponse).rejects.toMatchObject({response: { status: 400 } });
+        });
+    
+        it("When updating new oraganization with empty phone, Expect to return HTTP Status 400", async () => {
+          //Arrange
+          let oraganizationApi = new OrganizationApi(configurationWithAuthHeader);
+          //Act
+          let payload = TdeiObjectFaker.getOrganization();
```
Comment (MashB) [inline src/__tests__/organization.tests.ts]: Please look at the implementation in pathways for similar case

File: src/__tests__/organization.tests.ts [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+        console.log(organizationResponse);
+        expect(organizationResponse.data).toBeGreaterThan(0);
+  
+      });
+      });
+
+      describe("Validation", () => {
+        it("When updating new oraganization with empty name, Expect to return HTTP Status 400", async () => {
+          //Arrange
+          let oraganizationApi = new OrganizationApi(configurationWithAuthHeader);
```
Comment (MashB) [inline src/__tests__/organization.tests.ts]: Please look at the implementation in pathways for similar case

File: src/__tests__/organization.tests.ts [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+    
+          //Assert
+          await expect(oraganizationResponse).rejects.toMatchObject({response: { status: 400 } });
+        });
+    
+        it("When updating new oraganization with empty address, Expect to return HTTP Status 400", async () => {
+          //Arrange
+          let oraganizationApi = new OrganizationApi(configurationWithAuthHeader);
+          //Act
+          let payload = TdeiObjectFaker.getOrganization();
```
Comment (MashB) [inline src/__tests__/organization.tests.ts]: Please look at the implementation in pathways for similar case

File: src/__tests__/organization.tests.ts [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+    
+          //Assert
+          await expect(oraganizationResponse).rejects.toMatchObject({response: { status: 400 } });
+        });
+    
+        it('When updating new organization with invalid polygon, Expect to return HTTP Status 400', async () => {
+          //Arrange
+          let oraganizationApi = new OrganizationApi(configurationWithAuthHeader);
+          //Act
+          let payload = TdeiObjectFaker.getOrganization();
```
Comment (MashB) [inline src/__tests__/organization.tests.ts]: Please look at the implementation in pathways for similar case

File: src/__tests__/organization.tests.ts [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+    
+          const oraganizationResponse = oraganizationApi.updateOrganization(payload);
+    
+          //Assert
+          await expect(oraganizationResponse).rejects.toMatchObject({ response: { status: 400 } });
+         });
+        });
+      });
+
+      describe('Get Station', () => {
```
Comment (MashB) [inline src/__tests__/organization.tests.ts]: Wrong name

File: src/__tests__/organization.tests.ts [hunk 7]
```diff
@@ -0,0 +1,319 @@
...
+        describe('Auth', () => {
+          it("When no auth token provided, Expect to return HTTP status 401", async () => {
+            //Arrange
+            let oraganizationApi = new OrganizationApi(configurationWithoutAuthHeader);
+            //Act
+            const request = async () => {
```
Comment (MashB) [inline src/__tests__/organization.tests.ts]: Change the syntax as done previous cases

File: src/__tests__/organization.tests.ts [hunk 8]
```diff
@@ -0,0 +1,319 @@
...
+
+            //Act
+            const oraganizationResponse = await oraganizationApi.getOrganization(seederData?.organizationId, undefined, undefined, undefined, undefined);
+
+            const data = oraganizationResponse.data;
+
+            //Assert
+              expect(oraganizationResponse.status).toBe(200);
+              expect(Array.isArray(data)).toBe(true);
+              expect(data).toBeInstanceOf(Array);
```
Comment (MashB) [inline src/__tests__/organization.tests.ts]: Duplicate assertion

File: src/__tests__/organization.tests.ts [hunk 9]
```diff
@@ -0,0 +1,319 @@
...
+              expect(data[0].tdei_org_id).toEqual(seederData?.organizationId);
+            });
+
+            it('When searched with organization name filter, Expect to return list of organizations matching fiter', async () => {
+            //Arrange
+            let oraganizationApi = new OrganizationApi(configurationWithAuthHeader);
+
+            //Act
+            const oraganizationResponse = await oraganizationApi.getOrganization(undefined, seederData?.service?.service_name, undefined, undefined, undefined);
```
Comment (MashB) [inline src/__tests__/organization.tests.ts]: Why service name ?

File: src/__tests__/user-management.test.ts [hunk 10]
```diff
@@ -1,6 +1,7 @@
-import { TDEIROLES, Utility } from "../utils";
```
Comment (MashB) [inline src/__tests__/user-management.test.ts]: Please take latest and undo your chAnges to user management service test cases

File: src/__tests__/organization.tests.ts (hunk 1 above)
Comment (iAKM) [reply src/__tests__/organization.tests.ts]: fixed

File: src/__tests__/organization.tests.ts (hunk 2 above)
Comment (iAKM) [reply src/__tests__/organization.tests.ts]: fixed

File: src/__tests__/organization.tests.ts (hunk 3 above)
Comment (iAKM) [reply src/__tests__/organization.tests.ts]: fixed

File: src/__tests__/organization.tests.ts (hunk 4 above)
Comment (iAKM) [reply src/__tests__/organization.tests.ts]: fixed

File: src/__tests__/organization.tests.ts (hunk 5 above)
Comment (iAKM) [reply src/__tests__/organization.tests.ts]: fixed

File: src/__tests__/organization.tests.ts (hunk 6 above)
Comment (iAKM) [reply src/__tests__/organization.tests.ts]: fixed

File: src/__tests__/organization.tests.ts (hunk 7 above)
Comment (iAKM) [reply src/__tests__/organization.tests.ts]: fixed

File: src/__tests__/organization.tests.ts (hunk 8 above)
Comment (iAKM) [reply src/__tests__/organization.tests.ts]: assertion is similar to the one in pathways.

File: src/__tests__/organization.tests.ts (hunk 9 above)
Comment (iAKM) [reply src/__tests__/organization.tests.ts]: fixed

File: src/__tests__/user-management.test.ts (hunk 10 above)
Comment (iAKM) [reply src/__tests__/user-management.test.ts]: fixed

File: src/__tests__/organization.tests.ts [hunk 11]
```diff
@@ -0,0 +1,301 @@
...
+
+    describe("Functional", () => {
+      it("When updating new oraganization, Expect to return newly updated org id", async () => {
+        //Arrange
+        let oraganizationApi = new OrganizationApi(configurationWithAuthHeader);
+        //Act
+        const organizationResponse = await oraganizationApi.updateOrganization(TdeiObjectFaker.getOrganization());
+        //Assert
+        expect(organizationResponse.status).toBe(200);
+        console.log(organizationResponse);
```
Comment (susrisha) [inline src/__tests__/organization.tests.ts]: remove log

File: src/__tests__/organization.tests.ts (hunk 8 above)
Comment (susrisha) [reply src/__tests__/organization.tests.ts]: that is double assertion. Need to be removed in pathways as well.

File: src/__tests__/organization.tests.ts [hunk 12]
```diff
@@ -0,0 +1,301 @@
...
+              expect(data).toBeInstanceOf(Array);
+              expect(data[0].tdei_org_id).toEqual(seederData?.organizationId);
+            });
+
+            it('When searched with organization name filter, Expect to return list of organizations matching fiter', async () => {
+            //Arrange
+            let oraganizationApi = new OrganizationApi(configurationWithAuthHeader);
+
+            //Act
+            const oraganizationResponse = await oraganizationApi.getOrganization(undefined, seederData?.organization?.org_name, undefined, undefined, undefined);
```
Comment (susrisha) [inline src/__tests__/organization.tests.ts]: parameters after the org_name are not needed.

File: src/__tests__/organization.tests.ts [hunk 13]
```diff
@@ -0,0 +1,301 @@
...
+
+            const data = oraganizationResponse.data;
+
+            //Assert
+              expect(oraganizationResponse.status).toBe(200);
+              expect(Array.isArray(data)).toBe(true);
+              expect(data).toBeInstanceOf(Array);
```
Comment (susrisha) [inline src/__tests__/organization.tests.ts]: dual assertion

**Review by susrisha — CHANGES_REQUESTED** [review]

File: src/data.seed.ts [hunk 14]
```diff
@@ -16,6 +16,7 @@ export interface StationInterface {
 
 export class SeedDetails {
     organizationId: string | undefined;
```
Comment (priya-badmanji) [inline src/data.seed.ts]: remove this property , we can use Organization property defined below everywhere

File: src/data.seed.ts [hunk 15]
```diff
@@ -80,7 +81,8 @@ class SeedData {
         } else {
             try {
                 console.log("Generating seed data");
-                this.data.organizationId = await this.createOrganization();
+                this.data.organizationId = await this.createOrganizationId();
```
Comment (priya-badmanji) [inline src/data.seed.ts]: Remove this , we can use below property

File: src/data.seed.ts [hunk 16]
```diff
@@ -99,13 +101,22 @@ class SeedData {
         await writeFile('./seed.data.json', JSON.stringify(this.data), 'utf8');
     }
 
-    private async createOrganization(): Promise<string> {
+    private async createOrganizationId(): Promise<string> {
```
Comment (priya-badmanji) [inline src/data.seed.ts]: Delete this method, below createOrganization() is sufficient

File: src/__tests__/user-management.test.ts (hunk 10 above)
Comment (priya-badmanji) [reply src/__tests__/user-management.test.ts]: Still not, you are overriding all the implementations. Please check difference once. Take latest and undo any changes to user-management.test file

File: src/__tests__/organization.tests.ts (hunk 11 above)
Comment (iAKM) [reply src/__tests__/organization.tests.ts]: fixed

File: src/__tests__/organization.tests.ts (hunk 12 above)
Comment (iAKM) [reply src/__tests__/organization.tests.ts]: fixed

File: src/__tests__/organization.tests.ts (hunk 13 above)
Comment (iAKM) [reply src/__tests__/organization.tests.ts]: fixed

**Review by priya-badmanji — CHANGES_REQUESTED** [review]

File: src/data.seed.ts (hunk 14 above)
Comment (iAKM) [reply src/data.seed.ts]: fixed

Comment (iAKM) [reply src/data.seed.ts]: removing this property affected Pathways. Changed the values accordingly.

File: src/data.seed.ts (hunk 15 above)
Comment (iAKM) [reply src/data.seed.ts]: fixed

File: src/data.seed.ts (hunk 16 above)
Comment (iAKM) [reply src/data.seed.ts]: fixed

**Review by MashB — APPROVED** [review]

**Review by iAKM — COMMENTED** [review]
fixed

**Review by iAKM — COMMENTED** [review]
fixed

**Review by iAKM — COMMENTED** [review]
fixed

**Review by iAKM — COMMENTED** [review]
fixed

**Review by susrisha — APPROVED** [review]

**Review by susrisha — APPROVED** [review]

**Review by priya-badmanji — APPROVED** [review]

**Review by priya-badmanji — APPROVED** [review]

**Pull request merged**